api/api.go

```diff
@@ -82,6 +82,15 @@ const (
 	HTTPPartitionEnvName = "CONSUL_PARTITION"
 )
 
+type StatusError struct {
+	Code int
+	Body string
+}
+
+func (e StatusError) Error() string {
+	return fmt.Sprintf("Unexpected response code: %d (%s)", e.Code, e.Body)
```

Comment:
I'm on board with this but I see a potential problem with breaking existing string manipulations.

For one, in our testing code we do a fair amount of `expected: Unexpected response code: ...`. So doing that style change now will result in touching a number of file. Happy to address this in a subsequent PR.

That said, I'm a bit concerned for the community at large. For example, folks may be checking for the capitalized `U` like here: https://discuss.hashicorp.com/t/golang-consul-api-pulling-out-the-statuscode-from-deeply-nested-responses/18037. Then by virtue of making this change, we may _subtly_ break their code checks/ error handling.

What are your thoughts on the second point here?

Reply:
Ah ya, good call! I missed that this was an existing error message. Since there was no other way for users to check for error code previous, I think you're right we need to keep it as-is.